services/src/main/scala/cromwell/pipeline/service/ProjectFileService.scala

```diff
@@ -32,11 +33,20 @@ class ProjectFileService(
 
   def buildConfiguration(
     projectId: ProjectId,
-    projectFile: ProjectFile
-  ): Future[Either[ValidationError, ProjectConfiguration]] =
-    Future(womTool.inputsToList(projectFile.content.content)).map {
-      case Left(value) => Left(ValidationError(value.toList))
-      case Right(nodes) =>
-        Right(ProjectConfiguration(projectId, active = true, List(ProjectFileConfiguration(projectFile.path, nodes))))
+    projectFilePath: Path,
+    version: Option[PipelineVersion],
+    userId: UserId
+  ): Future[ProjectConfiguration] =
+    for {
+      project <- projectService.getUserProjectById(projectId, userId)
+      eitherFile <- projectVersioning.getFile(project, projectFilePath, version)
+    } yield eitherFile match {
+      case Right(file) =>
+        womTool.inputsToList(file.content.content) match {
+          case Right(nodes) =>
+            ProjectConfiguration(projectId, active = true, List(ProjectFileConfiguration(file.path, nodes)))
+          case Left(e) => throw ValidationException(e.toList.mkString(", "))
+        }
+      case Left(versioningException) => throw versioningException
```

Comment:
In Scala we don't throw exceptions. Especially when method's return type allows us to fail gracefully.

Reply:
Fixed.